src/interactivity/win32/Clipboard.cpp

```diff
@@ -127,16 +129,31 @@ void Clipboard::StringPaste(_In_reads_(cchData) const wchar_t* const pData,
 // Arguments:
 // - pData - the text to convert
 // - cchData - the size of pData, in wchars
+// - bracketedPaste - should this be bracketed with paste control sequences
 // Return Value:
 // - deque of KeyEvents that represent the string passed in
 // Note:
 // - will throw exception on error
 std::deque<std::unique_ptr<IInputEvent>> Clipboard::TextToKeyEvents(_In_reads_(cchData) const wchar_t* const pData,
-                                                                    const size_t cchData)
+                                                                    const size_t cchData,
+                                                                    const bool bracketedPaste)
 {
     THROW_HR_IF_NULL(E_INVALIDARG, pData);
 
     std::deque<std::unique_ptr<IInputEvent>> keyEvents;
+    const auto pushControlSequence = [&](const std::wstring_view sequence) {
+        std::for_each(sequence.begin(), sequence.end(), [&](const auto wch) {

```

Comment:
A range-for should be simpler I think.

Reply:
The main reason I used `std::for_each` was so I could avoid having to put the `{` on a new line, which I really loathe. I know that's petty. But I did also check out the generated assembly, and `std::for_each` actually seemed to be a little more efficient - at least fewer instructions - so I figured that might count as a more rational justification.

That said, it's not a big deal to change if you really don't like it this way.

Reply:
> K&R gang 4 life.

Absolutely!

> At least in my little test it produces the same assembly though: https://godbolt.org/z/crj8aM1qr

That is interesting. That looks nothing like the assembly I'm seeing generated in Visual Studio. I don't think my version of VS is that out of date (the C/C++ Compiler is 19.34.31937), but maybe it's just the other code in the method that alters the way those lambdas are optimized.

Not that I'm seeing a huge difference between the two anyway. I just found it interesting that the `std::for_each` seemed a little better. I was concerned it would be worse.

> Although you might be interested in using `std::ranges::for_each(sequence, ...)`

Yes, I was hoping there was something like that, so I'm glad you brought that up. Unfortunately it also introduces additional overhead in the generated assembly. For the most part it looks almost identical to the `std::for_each` version, but for some reason it adds a `security_cookie`, and an associated `security_check_cookie` call on exit. That's really annoying.

Reply:
On looking at your godbolt link again, I realised now why it didn't look anything like what I was expecting. The function immediately below `pushControlSequence$ = 96` (which is identical for both versions), is just the code that calls the `pushControlSequence` lambda. I think the lambda itself is further below that (following `sequence$ = 88` in the one source, and `sequence$ = 104` in the other).

In godbolt those too lambdas are massively different, but that's mostly because the one seems to have partially inlined the `push_back` calls. I don't get that inlining in VS for either of them, but the general structure is similar to what I'm seeing. The differences I see in VS are things like the one having more registers saved on the stack, and extra register loads in the body of the loop.